frontend/src/components/Main/ModelsForm/ModelsCheckBox.js

```diff
@@ -1,73 +1,734 @@
-                        {item.values.map((dev, index) => {
-                            return (
-                                <td key={index}>
-                                    <input
-                                        name={dev.name}
-                                        className="align-middle"
-                                        type="checkbox"
-                                        onChange={e =>
-                                            props.toggleModelsCheckBox(dev.name, e.target.checked)
-                                        }
-                                        checked={dev.isChecked}
-                                        disabled={dev.isDisabled}
-                                    />
+import * as mc from "../../../constants/mainConstants";
+
+const ModelsCheckBox = observer(props => {
+    const {store} = props,
+        isChecked = function(name, model) {
+            let checked = false;
+            if (name in store.models) {
+                if (name === mc.BAYESIAN_MODEL_AVERAGE) {
+                    checked = store.models[name].findIndex(obj => obj.model === model) > -1;
+                } else {
+                    checked = store.models[name].indexOf(model) > -1;
+                }
+            }
+            return checked;
+        },
+        getPriorWeightValue = function(model) {
+            let prior_weight = 0;
+            if (mc.BAYESIAN_MODEL_AVERAGE in store.models) {
+                let obj = store.models[mc.BAYESIAN_MODEL_AVERAGE].find(obj => obj.model === model);
 
-                                    {dev.name.includes(model.Bayesian_Model_Average) ? (
-                                        <input
-                                            className="form-control form-control-sm align-middle col-sm-3 pl-1 float-right"
-                                            type="number"
-                                            value={dev.prior_weight}
-                                            onChange={e =>
-                                                props.savePriorWeight(
-                                                    dev.name,
-                                                    parseFloat(e.target.value)
-                                                )
-                                            }
-                                        />
-                                    ) : null}
-                                </td>
-                            );
-                        })}
+                if (obj != undefined) {
+                    prior_weight = obj.prior_weight;
+                }
+            }
+            return prior_weight;
+        };
+    return (
+        <>
+            {store.getModelType === mc.MODEL_CONTINUOUS ? (
+                <tbody>
+                    <tr>
+                        <td className="text-left align-middle">Exponential</td>
+                        <td>
+                            <input
```

Comment:
so much duplication here, make a new component that returns the td we want. The goal here is to reduce the # lines of code

```javascript
<ModelCheckboxTd type="frequentist_restricted" model="Exponential"/>
```